src/components/input/StateEditor.ts

```diff
@@ -0,0 +1,51 @@
+import { State } from "./State"
+
+export class StateEditor implements State {
+	get value(): string { return this.state.value }
+	set value(value: string) { this.state.value = value }
```

Comment:
And not have any setters so that there is no manual bypassing of the logic to update the selection.